src/HotChocolate/ApolloFederation/src/ApolloFederation/TagDirectiveType.cs

```diff
@@ -0,0 +1,59 @@
+using HotChocolate.ApolloFederation.Constants;
+using HotChocolate.ApolloFederation.Properties;
+
+namespace HotChocolate.ApolloFederation;
+
+/// <summary>
+/// The @tag directive is used to applies arbitrary string
+/// metadata to a schema location. Custom tooling can use
+/// this metadata during any step of the schema delivery flow,
+/// including composition, static analysis, and documentation
+///
+/// <example>
+/// # extended from the Users service
+/// extend type User @key(fields: "id") {
+///   id: ID! @external
+///   email: String @tag(name: "public")
+///   customerNotes: String @tag(name: "internal")
+/// }
+/// </example>
+/// </summary>
+public sealed class TagDirective
+{
+    public string Name { get; set; } = default!;
```

Comment:
This should be immutable and use constructor injection of the value

Reply:
@michaelstaib updated as per your suggestion in comment below to make it immutable and value taken from constructor